### PR TITLE
New CoH dataset

### PR DESCRIPTION
**Pull request by andrewram4287**
Updating the CoH will new patients. Only using patients that have experiments from Untreated, 50mg IL6, IL10, IL2, IFNg, and IL4. Implemented regression model based on the factors. Created functions to plot factors, covariances, and raw data as well.

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
I need an idea of what the intended changes are here.

**Comment by andrewram4287** [comment]
> I need an idea of what the intended changes are here.

Wrote a description, let me know if further clarification is needed.

**Comment by aarmey** [comment]
Ok, there are a ton of changes here. Is there any way to split it up?

**Comment by andrewram4287** [comment]
> Ok, there are a ton of changes here. Is there any way to split it up?

Yeah i can do that if it'll make it easier to interpret. I can make more PR and separate out the changes if thats the easiest way to go about it

**Comment by aarmey** [comment]
@andrewram4287 please resolve conflicts here to start on this PR, then check that it builds.

File: gmm/tests/test_GMM.py [hunk 1]
```diff
@@ -25,6 +25,40 @@
...
+@pytest.mark.parametrize("rank", [3, 10])
+@pytest.mark.parametrize("nk_r", [True, False])
+def test_infer_rank(rank, nk_r):
+    """Test that we correctly infer the vector rank."""
+    x0 = vector_guess(meanShape, rank=rank, nk_rearrange=nk_r)
+    rank_inf = infer_rank(x0.size, meanShape, nk_rearrange=nk_r)
+    assert rank_inf == rank
+
+
+def test_cov_fit():
```
Comment (aarmey) [inline gmm/tests/test_GMM.py:37]: These are duplicated now due to failed merge.

File: gmm/tensor.py [hunk 2]
```diff
@@ -180,6 +181,18 @@ def get_factors_xarray(self, X):
                         coords={"Cmp": cmpCol, **coordinates})
 
         return da
+    
+    def reorient_factors(self, numb_fact):
```
Comment (aarmey) [inline gmm/tensor.py:185]: Is this relevant to the PR?

File: gmm/tensor.py [hunk 3]
```diff
@@ -1,17 +1,18 @@
+from re import S
 import numpy as np
 import pandas as pd
 import jax.numpy as jnp
 import jax.scipy.special as jsp
 import tensorly as tl
 from tqdm import tqdm
 import xarray as xa
+import jax.lax as jlx
```
Comment (aarmey) [inline gmm/tensor.py:9]: Unused.

File: gmm/tensor.py [hunk 4]
```diff
@@ -1,17 +1,18 @@
+from re import S
```
Comment (aarmey) [inline gmm/tensor.py:1]: Unused.

**Review by aarmey — CHANGES_REQUESTED** [review]
A few comments. Why are there so many changes to figures that don't involve the CoH data? Also, it would be helpful to add a unit test for the import functions.

**Pull request closed without merging**